### PR TITLE
ci: parametrized framework tests

### PR DESCRIPTION
**Pull request by gnufede**
CI: parametrize framework tests to run them separately with Profiling, IAST or Appsec enabled

## Checklist

- [x] Change(s) are motivated and described in the PR description
- [x] Testing strategy is described if automated tests are not included in the PR
- [x] Risks are described (performance impact, potential for breakage, maintainability)
- [x] Change is maintainable (easy to change, telemetry, documentation)
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed or label `changelog/no-changelog` is set
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/))
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))
- [x] If this PR changes the public interface, I've notified `@DataDog/apm-tees`.

## Reviewer Checklist

- [x] Title is accurate
- [x] All changes are related to the pull request's stated goal
- [x] Description motivates each change
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- [x] Testing strategy adequately addresses listed risks
- [x] Change is maintainable (easy to change, telemetry, documentation)
- [x] Release note makes sense to a user of the library
- [x] Author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

**Review by avara1986 — APPROVED** [review]

File: .github/workflows/test_frameworks.yml [hunk 1]
```diff
@@ -26,11 +26,28 @@ jobs:
...
+            iast: 0
+            appsec: 0
+          - suffix: IAST
+            profiling: 0
+            iast: 1
+            appsec: 0
+          - suffix: APPSEC
+            profiling: 0
+            iast: 0
+            appsec: 1
```
Comment (erikayasuda) [inline .github/workflows/test_frameworks.yml:43]: Could we factor out each of these components like this?
```
profiles:
  profiling: &profiling
    suffix: Profiling
    profiling: 1
    iast: 0
    appsec: 0

  iast: &iast
    suffix: IAST
    profiling: 0
    iast: 1
    appsec: 0

  appsec: &appsec
    suffix: APPSEC
    profiling: 0
    iast: 0
    appsec: 1
```
And then use them like this?
```suggestion
# Reference the reusable items using aliases
strategy:
  matrix:
    include:
          - <<: *profiling
          - <<: *iast
          - <<: *appsec
```

**Review by erikayasuda — APPROVED** [review]
One non-blocker, otherwise LGTM

File: .github/workflows/test_frameworks.yml (hunk 1 above)
Comment (gnufede) [reply .github/workflows/test_frameworks.yml:43]: We could, if only github actions would support yaml anchors: https://github.com/actions/runner/issues/1182

Comment (erikayasuda) [reply .github/workflows/test_frameworks.yml:43]: Ugh that's so tragic... 😞 It's been open for several _years_

**Pull request merged**